client/my-sites/sharing/connections/connection.jsx

```diff
@@ -1,159 +1,175 @@
-		onDisconnect: React.PropTypes.func,              // Handler to invoke when disconnecting
-		isDisconnecting: React.PropTypes.bool,           // Is a service disconnection request pending?
-		showDisconnect: React.PropTypes.bool,            // Display an inline disconnect button
-		onRefresh: React.PropTypes.func,                 // Handler to invoke when refreshing
-		isRefreshing: React.PropTypes.bool,              // Is a service refresh request pending?
-		onToggleSitewideConnection: React.PropTypes.func // Handler to invoke when toggling sitewide connection
-	},
-
-	getInitialState: function() {
-		return { isSavingSitewide: false };
-	},
-
-	componentDidUpdate: function( prevProps ) {
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getSelectedSite } from 'state/ui/selectors';
+import { getUser } from 'state/users/selectors';
+import { recordGoogleEvent } from 'state/analytics/actions';
+
+class SharingConnection extends Component {
+	static propTypes = {
+		site: PropTypes.object,                    // The site for which the connection was created
+		user: PropTypes.object,                    // A user object
+		connection: PropTypes.object.isRequired,   // The single connection object
+		service: PropTypes.object.isRequired,      // The service to which the connection is made
+		onDisconnect: PropTypes.func,              // Handler to invoke when disconnecting
+		isDisconnecting: PropTypes.bool,           // Is a service disconnection request pending?
+		showDisconnect: PropTypes.bool,            // Display an inline disconnect button
+		onRefresh: PropTypes.func,                 // Handler to invoke when refreshing
+		isRefreshing: PropTypes.bool,              // Is a service refresh request pending?
+		onToggleSitewideConnection: PropTypes.func // Handler to invoke when toggling sitewide connection
+	};
+
+	static defaultProps = {
+		isDisconnecting: false,
+		isRefreshing: false,
+		onDisconnect: () => {},
+		onRefresh: () => {},
+		onToggleSitewideConnection: () => {},
+		showDisconnect: false,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			isSavingSitewide: false
+		};
+
+		this.toggleSitewideConnection = this.toggleSitewideConnection.bind( this );
+	}
+
+	componentDidUpdate( prevProps ) {
 		if ( this.state.isSavingSitewide && this.props.connection.shared !== prevProps.connection.shared ) {
 			this.setState( { isSavingSitewide: false } );
 		}
-	},
-
-	getDefaultProps: function() {
-		return {
-			onDisconnect: function() {},
-			isDisconnecting: false,
-			showDisconnect: false,
-			onRefresh: function() {},
-			isRefreshing: false,
-			onToggleSitewideConnection: function() {}
-		};
-	},
+	}
 
-	disconnect: function() {
+	disconnect() {
 		if ( ! this.props.isDisconnecting ) {
 			this.props.onDisconnect( this.props.connection );
 		}
-	},
+	}
 
-	refresh: function() {
+	refresh() {
```

Comment:
I believe this needs to be bound to the instance, since it's a callback from `getReconnectButton`'s `<a />` tag and references `this`.